python/cudf/cudf/core/index.py

```diff
@@ -2703,6 +2706,144 @@ def categories(self):
+    freq: int = None,
+    closed: str = "right",
+    name: str = None,
+) -> "IntervalIndex":
+    """
+    Returns a fixed frequency IntervalIndex.
+
+    Parameters
+    start : numeric, default None
+    Left bound for generating intervals.
+
+    end : numeric , default None
+    Right bound for generating intervals.
+
+    freq : numeric, default None
+    The length of each interval. Must be consistent
+    with the type of start and end
+
+    name : str, default None
+    Name of the resulting IntervalIndex.
+
+    closed : {‘left’, ‘right’, ‘both’, ‘neither’}, default ‘right’
+    Whether the intervals are closed on the left-side, right-side,
+    both or neither.
+
+    Returns
+    IntervalIndex
+
+    Examples
+    --------
+    >>> import cudf
+    >>> import pandas as pd
+    >>> cudf.interval_range(start=0,end=5)
+    ... IntervalIndex([(0, 0], (1, 1], (2, 2], (3, 3], (4, 4], (5, 5]],
+              closed='right',dtype='interval')
+    """
+    if freq and periods and start and end:
+        raise ValueError(
+            "Of the four parameters: start, end, periods, and "
+            "freq, exactly three must be specified"
+        )
+    elif periods and not freq:
+        if end is not None:
+            end = end + 1
+        periods_array = cupy.asarray(cupy.arange(start, end))
+        hist, bin_edges = cupy.histogram(periods_array, periods)
+        # cupy.histogram turns all arrays into a float array
+        # this can cause the dtype to be a float instead of an int
+        # the below adjusts for this
+        if cupy.all(cupy.mod(bin_edges, 1) == 0):
+            bin_edges = bin_edges.astype(int)
+        left_col = bin_edges[:-1]
+        right_col = bin_edges[1:]
+    else:
+        if freq and periods and end:
+            start = end - (freq * periods)
+        if freq and periods and start:
+            end = freq * periods + start
+        left_col = cupy.arange(start, end, freq)
+        if end is not None:
+            end = end + 1
+        if start is not None and freq is not None:
+            new_freq = start + freq
+        elif start is not None and freq is None:
+            new_freq = start
+        right_col = cupy.arange(new_freq, end, freq)
+        if len(left_col) != len(right_col):
+            if end and freq is not None:
+                end = end - freq
+            left_col = cupy.arange(start, end, freq)
+        if len(right_col) == 0 or len(left_col) == 0:
+            return pd.IntervalIndex([], closed=closed)
+
+    interval_col = column.build_interval_column(
+        left_col, right_col, closed=closed
+    )
+    return IntervalIndex(interval_col)
+
+
+class IntervalIndex(GenericIndex):
+    """
+    Immutable index of intervals that are closed on the same side.
+
+    Parameters
+    data : array-like (1-dimensional)
+    Array-like containing Interval objects from which to build the
+    IntervalIndex.
+
+    closed : {‘left’, ‘right’, ‘both’, ‘neither’}, default ‘right’
```

Comment:
Am I seeing the unicode quotation mark here too? Similar issue:
https://github.com/rapidsai/cudf/pull/6883#pullrequestreview-568725976

Reply:
Thanks, Michael, generally need to pay more attention to this 😆